src/System.Data.Common/src/System/Data/Common/DbProviderFactory.cs

```diff
@@ -19,15 +19,9 @@ public virtual bool CanCreateDataAdapter
             {
                 if (!_canCreateDataAdapter.HasValue)
                 {
-                    var adapter = CreateDataAdapter();
-                    if (adapter == null)
+                    using (var adapter = CreateDataAdapter())
                     {
-                        _canCreateDataAdapter = false;
-                    }
-                    else
-                    {
-                        _canCreateDataAdapter = true;
-                        adapter.Dispose();
+                        _canCreateDataAdapter = adapter is object;
```

Comment:
We've generally stayed away from this form of not-null check, as it's not obvious to most readers what it's actually verifying.  Please change it to `adapter != null`.

Reply:
I did wonder about that; will change. Thanks.